regtest/README.md

```diff
@@ -0,0 +1,183 @@
+```
+
+As soon as the `PREIMAGE_REVEALED` message appears, you know the HTLC was
+published to the chain. Finish the process by mining yet another block (in the
+second terminal window):
+
+```shell
+$ ./regtest.sh mine 1
+```
+
+Your `monitor` window should now show the completion of the swap:
+
+```shell
+2021-03-19T17:01:39Z LOOP_OUT INITIATED 0.005 BTC - bcrt1qwgzsmag0579twr4d8hgfvarqxn27mdmdgfh7wc2hdlupsfqtxwyq5zltqn
+2021-03-19T17:04:37Z LOOP_OUT PREIMAGE_REVEALED 0.005 BTC - bcrt1qwgzsmag0579twr4d8hgfvarqxn27mdmdgfh7wc2hdlupsfqtxwyq5zltqn
+2021-03-19T17:06:25Z LOOP_OUT SUCCESS 0.005 BTC - bcrt1qwgzsmag0579twr4d8hgfvarqxn27mdmdgfh7wc2hdlupsfqtxwyq5zltqn (cost: server 50, onchain 6687, offchain 0)
+```
+
+Congratulations, you just did a loop out!
+
+# Loop in
+
+Looping in is very similar but does the opposite: swap your on-chain balance for
+off-chain (channel) balance:
+
+```shell
+$ ./regtest.sh loop in --amt 500000
+Send on-chain:                             500000 sat
+Receive off-chain:                         492279 sat
+Estimated total fee:                         7721 sat
+
+CONTINUE SWAP? (y/n): y
+Swap initiated
+ID:           d95167e33e79df292e5ffeda29d34614a5a087c9378ffae003b9a25bb0f1c2e4
+HTLC address (P2WSH): bcrt1q4p0pead4tfepm683fff8fl8g3kpx8spjztuquu2sctfzed9rufls2z0g2f
+
+Run `loop monitor` to monitor progress.
+
+
+$ ./regtest.sh loop monitor
+Note: offchain cost may report as 0 after loopd restart during swap
+2021-03-19T17:17:13Z LOOP_IN HTLC_PUBLISHED 0.005 BTC - P2WSH: bcrt1q4p0pead4tfepm683fff8fl8g3kpx8spjztuquu2sctfzed9rufls2z0g2f
+```
+
+In your second terminal window, go ahead and mine a block:
+
+```shell
+$ ./regtest.sh mine 1
+```
+
+You should see your previous window with the `monitor` command update:
+
+```shell
+2021-03-19T17:17:13Z LOOP_IN HTLC_PUBLISHED 0.005 BTC - P2WSH: bcrt1q4p0pead4tfepm683fff8fl8g3kpx8spjztuquu2sctfzed9rufls2z0g2f
+2021-03-19T17:18:38Z LOOP_IN INVOICE_SETTLED 0.005 BTC - P2WSH: bcrt1q4p0pead4tfepm683fff8fl8g3kpx8spjztuquu2sctfzed9rufls2z0g2f (cost: server -499929, onchain 7650, offchain 0)
+```
+
+After mining yet another block, the process should complete:
+
+```shell
+$ ./regtest.sh mine 1
+```
+
+```shell
+2021-03-19T17:17:13Z LOOP_IN HTLC_PUBLISHED 0.005 BTC - P2WSH: bcrt1q4p0pead4tfepm683fff8fl8g3kpx8spjztuquu2sctfzed9rufls2z0g2f
+2021-03-19T17:18:38Z LOOP_IN INVOICE_SETTLED 0.005 BTC - P2WSH: bcrt1q4p0pead4tfepm683fff8fl8g3kpx8spjztuquu2sctfzed9rufls2z0g2f (cost: server -499929, onchain 7650, offchain 0)
+2021-03-19T17:19:21Z LOOP_IN SUCCESS 0.005 BTC - P2WSH: bcrt1q4p0pead4tfepm683fff8fl8g3kpx8spjztuquu2sctfzed9rufls2z0g2f (cost: server 71, onchain 7650, offchain 0)
+```
+
+# Using the Loop server in an existing setup
+
+This `docker-compose` is only meant as a demo and quick start help. You can of
+course also integrate the Loop server Docker image into your existing `regtest`
+environment.
```

Comment:
make sure to use different lnds for server and client because you can't pay to yourself